src/server/page_server.cpp

```diff
@@ -515,7 +517,11 @@ page_server::set_passive_tran_server_connection (cubcomm::channel &&chn)
   er_log_debug (ARG_FILE_LINE, "Passive transaction server connected to this page server. Channel id: %s.\n",
 		channel_id.c_str ());
 
-  m_passive_tran_server_conn.emplace_back (new connection_handler (chn, transaction_server_type::PASSIVE, *this));
+  {
+    std::lock_guard lk_guard (m_conn_mutex);
```

Comment:
Nit.
For the functions `set_active_tran_server_connection` and `set_passive_tran_server_connection` the mutex would actually not be needed because these functions are executed both on the master connection loop (`css_master_thread`). The usage of the mutex is to synchronize with the disconnects which can happen more asynchronously.

Maybe it's useful to add this comment somewhere.

Reply:
My comment is not negative.
It was only an explanation for why it is needed to guard both `set_active_tran_server_connection` and `set_passive_tran_server_connection` together with the disconnect functionality even if these two functions are called from the same thread.

The idea was to draw a comment in the code from my comment:
```
// Even if the functions set_active_tran_server_connection and set_passive_tran_server_connection are
// called from the same thread (master-server connection handler) the mutex is actually needed.
// The usage of the mutex is to synchronize with the disconnects which are trigered from each connection
// handler's connection threads (inbound or outbound).
```

Reply:
Aha, OK. I got it wrong. I'll add the comment in both functions.